support-frontend/app/controllers/CreateSubscriptionController.scala

```diff
@@ -36,63 +44,52 @@ class CreateSubscriptionController(
-          case None =>
-            SafeLogger.info(s"Guest user ${request.body.email} is attempting to create a new ${request.body.product.describe} [${request.uuid}]")
-            createGuestUserAndHandleRequest().getOrElse(InternalServerError)
-        }
-    })
-
-  private def createGuestUserAndHandleRequest()(implicit request: OptionalAuthRequest[CreateSupportWorkersRequest]): EitherT[Future, String, Result] =
-    for {
-      userId <- identityService.getOrCreateUserIdFromEmail(request.body.email, request.body.firstName, request.body.lastName)
-      result <- EitherT.right[String](
-        handleCreateSupportWorkersRequest(IdMinimalUser(userId, None))
-      )
-    } yield result
+    LoggingAndAlarmOnFailure {
+      MaybeAuthenticatedAction.async(circe.json[CreateSupportWorkersRequest]) { implicit request =>
+
+        val maybeLoggedInIdentityIdAndEmail = request.user.map(authIdUser => IdentityIdAndEmail(authIdUser.id, authIdUser.primaryEmailAddress))
+        logIncomingRequest(request, maybeLoggedInIdentityIdAndEmail)
+
+        val errorOrStatusResponse = for {
+          userAndEmail <- maybeLoggedInIdentityIdAndEmail match {
+            case Some(identityIdAndEmail) => EitherT.pure[Future, CreateSubscriptionError](identityIdAndEmail)
+            case None => getOrCreateIdentityUser(request.body).leftMap(ServerError)
+          }
+          _ <- validate(request)
+          supportWorkersUser = buildSupportWorkersUser(userAndEmail, request.body, testUsers.isTestUser(request))
+          statusResponse <- client.createSubscription(request, supportWorkersUser, request.uuid).leftMap[CreateSubscriptionError](ServerError)
+        } yield statusResponse
 
-  private def handleCreateSupportWorkersRequest(idMinimalUser: IdMinimalUser)(
-    implicit request: OptionalAuthRequest[CreateSupportWorkersRequest]
-  ): Future[Result] = {
+        toHttpResponse(errorOrStatusResponse, request.body.product)
 
-    val normalisedTelephoneNumber = NormalisedTelephoneNumber.fromStringAndCountry(request.body.telephoneNumber, request.body.billingAddress.country)
+      }
+    }
 
-    val createSupportWorkersRequest = request.body.copy(
-      telephoneNumber = normalisedTelephoneNumber.map(asFormattedString)
-    )
+  private def logIncomingRequest(request: OptionalAuthRequest[CreateSupportWorkersRequest], maybeLoggedInIdentityIdAndEmail: Option[IdentityIdAndEmail]) = {
+    val userDesc = maybeLoggedInIdentityIdAndEmail match {
+      case None => s"Guest User ${request.body.email}"
+      case Some(idAndEmail) => s"User ${idAndEmail.primaryEmailAddress}"
+    }
+    SafeLogger.info(s"$userDesc is attempting to create a new ${request.body.product.describe} [${request.uuid}]")
+  }
 
-    CheckoutValidationRules.validate(createSupportWorkersRequest) match {
-      case Valid =>
-        val result = for {
-          user <- identityService.getUser(idMinimalUser).leftMap(ServerError)
```

Comment:
we don't need to getUser any more as we can reuse the one retrieved as part of auth